### ISSUE TITLE
Would it be possible to bring back the -src flag ?

### ISSUE THREAD

**Issue by Croco-byte** [OWASP/Amass#959]
The `-src` flag was really useful, especially for integration with other tools. It seems like this feature was removed from the `enum` subcommand in latest releases, but was pertinent for various use cases.

There does not seem to be currently any way to print the sources of the enumerated subdomains.

Would it be possible to bring the `-src` flag back, or an equivalent ?

Thank you!

**Comment by rynmrtn** [comment]
@Croco-byte Thanks for opening this issue and highlighting the value of the source information in your workflow. We have heard from several others that this is also valuable to them.

When our team of contributors re-imagined and developed an entirely new model for representing and storing assets, source was not included in the first iteration that was integrated into amass. The good news is that we have plans to support data provenance (see [W3C](https://www.w3.org/2005/Incubator/prov/wiki/What_Is_Provenance), [owasp-amass/open-asset-model for details](https://github.com/owasp-amass/open-asset-model/blob/master/docs/taxonomy.md#source--data-provenance)) in upcoming releases to meet this particular need (and more).

While I have you (and any others that view this issue), can you please share how the source field proved valuable to the work you are doing and how you previously utilized it? This will help the the contributors understand the implications of future design decisions related to this topic.

**Comment by Croco-byte** [comment]
Thank you for the quick answer. The fact that data provenance should be back in upcoming releases is great news, I'm looking forward to it!

To provide a bit more context as requested, I am currently using amass as part of a wider enumeration / reconnaissance tool designed for pentesters. When gathering information about a target, one critical aspect is to sort the retrieved data (for instance, enumerated domains) to quickly identify reliable results, while leaving false positives aside. In order to do so, two key elements are taken into account as part of the tool's workflow: 1. The sources from which the data originates (a particular source may be more or less reliable) ; 2. The number of distinct sources from which the data originates (the greater the number of distinct sources return the same result, the more reliable that result should be).

In a nutshell, when it comes to enumeration tools, the data itself is of course very important, but providing as many indicators as possible in order to allow sorting this data is also a key feature in my opinion.

In any case, thanks a lot for the work and the time invested into amass, which is really a great tool!

**Comment by Croco-byte** [comment]
Hi, were there any progress made on this issue / can we still expect to see the `-src` flag make a comeback ?

Thanks!
